core/types.go

```diff
@@ -260,14 +260,38 @@ type RenewalInfo struct {
 // using a very simple renewal calculation: calculate a point 2/3rds of the way
 // through the validity period, then give a 2-day window around that. Both the
 // `issued` and `expires` timestamps are expected to be UTC.
-func RenewalInfoSimple(issued time.Time, expires time.Time) *RenewalInfo {
+func RenewalInfoSimple(issued time.Time, expires time.Time, now time.Time) *RenewalInfo {
 	validity := expires.Add(time.Second).Sub(issued)
 	renewalOffset := validity / time.Duration(3)
 	idealRenewal := expires.Add(-renewalOffset)
+	windowStart := idealRenewal.Add(-24 * time.Hour)
+	windowEnd := idealRenewal.Add(24 * time.Hour)
+
+	if expires.Before(now) {
+		// If the cert has expired, return `RenewalInfoImmediate` instead.
+		return RenewalInfoImmediate(now)
+	}
+	
+	// Ensure RenewalWindow is not after the expiry
+	if windowEnd.After(expires){
+		windowEnd = expires
+	}
+	// Ensure correct start for future issueds
+	if windowStart.After(issued){
```

Comment:
I didn't suggest this to handle issued timestamps that are in the future. I suggested this to handle the case where the certificate has a validity period of less than 48 hours, so that subtracting 24 hours from a point in the middle of the validity period results in a windowStart that precedes the issued timestamp.

Also, you've reversed the logic from what I suggested. This should be `if windowStart.Before(issued) {`.

Reply:
Sorry, I was trying to do this in one window while debugging another upstream project in the cloud.